sonic-ycabled/ycable/ycable_utilities/y_cable_helper.py

```diff
@@ -1664,6 +1674,154 @@ def get_muxcable_static_info_without_presence():
 
     return mux_info_static_dict
 
+def parse_grpc_response_link_and_oper_state(ret, response, read_side, query_type, port):
+    self_state = peer_state = 'unknown'
+
+    if ret is True and response is not None:
+        if len(response.portid) == 2 and len(response.state) == 2:
+            if int(read_side) == 0:
+                if response.state[0] == True:
+                    self_state = 'up'
+                elif response.state[0] == False:
+                    self_state = 'down'
+                # No other values expected, should we raise exception/msg
+                # TODO handle other responses
+                if response.state[1] == True:
+                    peer_state = 'up'
+                elif response.state[1] == False:
+                    peer_state = 'down'
+
+            elif int(read_side) == 1:
+                if response.state[1] == True:
+                    self_state = 'up'
+                elif response.state[1] == False:
+                    self_state = 'down'
+                if response.state[0] == True:
+                    peer_state = 'up'
+                elif response.state[0] == False:
+                    peer_state = 'down'
+
+        else:
+            helper_logger.log_warning("recieved an error port list while parsing response {} port state list size 0 {} {}".format(query_type, len(response.portid), len(response.state)))
+            self_state = 'unknown'
+            peer_state = 'unknown'
+    else:
+        self_state = 'unknown'
+        peer_state = 'unknown'
+
+    stat = grpc_port_stats.get(port,None)
+    if stat is not None:
+
+        if query_type == "link_state":
+            grpc_port_stats[port]["link_state_probe_count"] = str(int(stat.get("link_state_probe_count", 0)) + 1 )
+            grpc_port_stats[port]["peer_link_state_probe_count"] = str(int(stat.get("peer_link_state_probe_count", 0)) + 1 )
+        elif query_type == "oper_state":
+            grpc_port_stats[port]["operation_state_probe_count"] = str(int(stat.get("operation_state_probe_count", 0)) + 1 )
+            grpc_port_stats[port]["peer_operation_state_probe_count"] = str(int(stat.get("peer_operation_state_probe_count", 0)) + 1 )
```

Comment:
Just being curious, what are the counters used for?

Reply:
These counters are for noting how many time gRPC API's are being called from ycabled's perspective.